app/helpers/user_journey_helper.rb

```diff
@@ -26,4 +26,17 @@ def certificate_status(certificate)
       "IN USE"
     end
   end
+
+  def certificate_expiry_count(msa_components, sp_components)
+    all_components = msa_components + sp_components
+    expiring_certificates = 0
+    all_components.each do |component|
+      component.certificates.each do |certificate|
+        if certificate.expires_soon?
+          expiring_certificates += 1
+        end
+      end
+    end
+    expiring_certificates
+  end
```

Comment:
```suggestion
  def expiring_certificates_count(msa_components, sp_components)
    (msa_components + sp_components).map(:&certificates).map { |c| c.expires_soon? }.count
  end
```

just a thought 😛

Reply:
Get undefined method `expires_soon?`

Reply:
My bad. the initial map will bring back an empty array for any components that don't have any certs.

```
(msa_components + sp_components).map(&:certificates).flatten.select { |c| c.expires_soon? }.count
```

That'd work.